File: Iteration 1/code/CSE3063GRP12/src/core/general_providers/InstanceManager.java
package core.general_providers;
import core.database.abstracts.DatabaseManager;
import core.database.concretes.JsonManager;

public class InstanceManager {
    private static InstanceManager instance;

    private InstanceManager() {
        
    }

    public static InstanceManager getInstance() {
        if (instance == null) {
            instance = new InstanceManager();
        }
        return instance;
    }

    public DatabaseManager getDataBaseInstance() {
        
        return new JsonManager();
    }
}

File: Iteration3/code/CSE3063GRP12/src/core/general_providers/InstanceManager.py
from core.database.concretes.JsonManager import JsonManager

class InstanceManager:
    _instance = None
    _database_manager = None

    def __new__(cls):
        if not cls._instance:
            cls._instance = super(InstanceManager, cls).__new__(cls)
        return cls._instance

    def get_database_instance(self):
        if self._database_manager is None:
            self._database_manager = JsonManager()
        return self._database_manager